lib/process-services/src/lib/task-list/components/task-form/task-form.component.ts

```diff
@@ -272,15 +269,26 @@ export class TaskFormComponent implements OnInit {
   }
 
   isReadOnlyForm(): boolean {
-      return this.internalReadOnlyForm || !(this.isAssignedToMe() || this.canInitiatorComplete());
+    let readOnlyForm: boolean;
+    if (this.isCandidateMember()) {
+      readOnlyForm = this.internalReadOnlyForm || !this.isAssignedToMe();
+    } else {
+      readOnlyForm = this.internalReadOnlyForm || !(this.isAssignedToMe() || (this.canInitiatorComplete() && this.isProcessInitiator()));
+    }
+
+    return readOnlyForm;
+  }
+
+  isProcessInitiator(): boolean {
+    return this.currentLoggedUser && ( this.currentLoggedUser.id === +this.taskDetails.processInstanceStartUserId);
```

Comment:
+this.taskDetails.processInstanceStartUserId is something wanted?

Reply:
yeah we need it, where UserRepresentation id is a number type and taskDetails.processInstanceStartUserId is string type.